CHANGELOG.md

```diff
@@ -9,6 +9,12 @@ and this project adheres to
 ## Unreleased
 
 #### Breaking Changes
+- Unreachable code is now an error
+  - [#4587](https://github.com/bpftrace/bpftrace/pull/4587)
+- `return` is no longer allowed in `while` loops
```

Comment:
I understand the motivation for this change but at the same time this seems a bit restrictive to me. It means users can't add some conditional inside a loop that returns from the probe (they have to add a break and some external variable). Currently does returning from inside a loop cause some other side effects or UB?

Reply:
I've done an overhaul, and now support the existing semantics accurately.